### PR TITLE
fix(config): Enhance ReactConfig and improve file handling logic

### PR DESCRIPTION
**Pull request by thelooter**
Added new properties to ReactConfig, including `iconLibrary` and extended aliases with `lib` and `hooks`. These have been added by the 2.0 CLI.

 Improved filtering of `package.json` files to exclude hidden directories and added logic to remove block comments in Source parsing.

Implements changes first made in #61

### CODE REVIEW

File: src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/backend/sources/Source.kt [hunk 1]
```diff
@@ -111,6 +111,8 @@ abstract class Source<C : Config>(val project: Project, private val serializer:
             .split("\n").joinToString("\n") { line ->
                 // Remove // comments
                 line.substringBefore("//").trim()
+                // Remove /* */ comments
+                line.replace(Regex("/\\*.*?\\*/"), "").trim()
```
Comment (WarningImHack3r) [inline src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/backend/sources/Source.kt:115]: This sounds like a good idea but it causes issues
See [29e4726](https://github.com/WarningImHack3r/intellij-shadcn-plugin/commit/29e472674143797da1354f30e97e01023ad48d8e) and #66

File: src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/ui/ISPPanelPopulator.kt [hunk 2]
```diff
@@ -23,7 +23,8 @@ class ISPPanelPopulator(private val project: Project) {
         CoroutineScope(SupervisorJob() + Dispatchers.Default).async {
             return@async Pair(
                 SourceScanner.findShadcnImplementation(project),
-                FileManager.getInstance(project).getVirtualFilesByName("package.json").size
+                FileManager.getInstance(project).getVirtualFilesByName("package.json")
+                    .filter { !it.parent.name.startsWith('.') }.size
```
Comment (WarningImHack3r) [inline src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/ui/ISPPanelPopulator.kt:27]: You can add this directly inside the `getVirtualFilesByName` function instead

**Review by WarningImHack3r — CHANGES_REQUESTED** [review]
Thanks! Isn't there a way to also allow additional arbitrary values inside the deserializer?

File: src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/backend/sources/Source.kt (hunk 1 above)
Comment (thelooter) [reply src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/backend/sources/Source.kt:115]: Alright, I'll look into a more robust solution for this tomorrow

File: src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/ui/ISPPanelPopulator.kt (hunk 2 above)
Comment (thelooter) [reply src/main/kotlin/com/github/warningimhack3r/intellijshadcnplugin/ui/ISPPanelPopulator.kt:27]: I had an issue where it detected the file in `.next`. This fixed it

**Review by WarningImHack3r — APPROVED** [review]

**Comment by WarningImHack3r** [comment]
Thank you!!

**Comment by WarningImHack3r** [comment]
Can you check if there is any other upgrades needed to other frameworks configs and possibly make the parsers allow additional keys?

**Comment by thelooter** [comment]
Will do

**Comment by thelooter** [comment]
From what I can tell this was the only change. I couldn't find anything about the `component.json` file in the docs of https://github.com/stefan-karger/solid-ui, but if I interpret https://github.com/stefan-karger/solid-ui/blob/main/packages/cli/src/utils/config.ts correctly, no changes are necessary either.

**Comment by thelooter** [comment]
From my point of view, there is nothing standing in a way of a patch release

**Comment by WarningImHack3r** [comment]
> From my point of view, there is nothing standing in a way of a patch release

Good to know, thanks for the check! Can you, however, look into permissive parser(s) so there's never any incompatibility preventing the plugin from working? It must require existing keys (that are mandatory) but allow additional ones as well

**Comment by thelooter** [comment]
Do you want every Json Parser instance to ignore unknown keys, or only the ones that parse the component.json files?

**Comment by WarningImHack3r** [comment]
> Do you want every Json Parser instance to ignore unknown keys, or only the ones that parse the component.json files?

All of them, I see you already figured it out in #72!

**Pull request merged**